examples/all-clusters-app/esp32/CMakeLists.txt

```diff
@@ -32,7 +32,7 @@ if(${IDF_TARGET} STREQUAL "esp32")
 endif()
 
 project(chip-all-clusters-app)
-idf_build_set_property(CXX_COMPILE_OPTIONS "-std=gnu++17;-Os;-DLWIP_IPV6_SCOPES=0;-DCHIP_HAVE_CONFIG_H" APPEND)
+idf_build_set_property(CXX_COMPILE_OPTIONS "-std=gnu++17;-Os;-Wno-maybe-uninitialized;-DLWIP_IPV6_SCOPES=0;-DCHIP_HAVE_CONFIG_H" APPEND)
```

Comment:
If there's just e.g. one skip of uninitialized variable, might be easy to fix, instead of adding warning?

Reply:
There is no easy fix for this one,  DataModel::Nullable<chip::NodeId> adminNodeID; has already been initialized, but M5 compiler still confuse, here is two disable this warning as error since we know here is wrong

Reply:
The question is: can we disable this via pragma in the Optional copy constructor instead?

Reply:
In particular, this seems to work for me:
```
    constexpr Optional(const Optional & other) : mHasValue(other.mHasValue)
    {
        if (mHasValue)
        {
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
            new (&mValue.mData) T(other.mValue.mData);
#pragma GCC diagnostic pop
        }
    }
```

Reply:
You _might_ also be able to use the move ctor (by using `std::move` on the relevant bits in the initializer), but that may have the same problem....

Reply:
Yes, we can disable this warning via pragma in the Optional copy constructor, this will apply to all platforms instead of just the compiler used on esp32

Reply:
Let's do that then.

Reply:
Darwin does not know warning group '-Wmaybe-uninitialized'

/Users/runner/work/connectedhomeip/connectedhomeip/src/lib/core/Optional.h:72:32: error: unknown warning group '-Wmaybe-uninitialized', ignored [-Werror,-Wunknown-warning-option]
#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"

Reply:
This works for me both for the Xcode Darwin build and the esp32 build:
```
    constexpr Optional(const Optional & other) : mHasValue(other.mHasValue)
    {
        if (mHasValue)
        {
            // The -Wmaybe-uninitialized warning gets confused about the fact
            // that other.mValue is always initialized if other.mHasValue is
            // true, so suppress it for our access to other.mValue.
#pragma GCC diagnostic push
#if !defined(__clang__)
#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
#endif // defined(__GNUC__)
            new (&mValue.mData) T(other.mValue.mData);
#pragma GCC diagnostic pop
        }
    }
```

Reply:
@bzbarsky-apple 

Yes, I already verified this in CI,  the CI has already been green now, any left?

Reply:
Applied your suggested change, it is same as the one I already tested, just add some comments, so it should pass the CI as well